torch/optim/radam.py

```diff
@@ -379,22 +378,23 @@ def _multi_tensor_radam(
         del grouped_grads
 
         rect = [
-            _dispatch_sqrt(
-                (rho_t - 4)
-                * (rho_t - 2)
-                * rho_inf
-                / ((rho_inf - 4) * (rho_inf - 2) * rho_t)
-            )
-            if rho_t > 5
-            else 0
-            for rho_t in rho_t_list
+            torch.where(
+                torch.tensor(rho_t > 5),
+                _dispatch_sqrt(_dispatch_abs(
+                    (rho_t - 4)
+                    * (rho_t - 2)
+                    * rho_inf
+                    / ((rho_inf - 4) * (rho_inf - 2) * rho_t)
+                )),
+                0.0
+            ) for rho_t in rho_t_list
         ]
-        unrectified = [0 if rect > 0 else 1.0 for rect in rect]
+        unrectified = [torch.where(r > 0, 0.0, 1.0) for r in rect]
 
         bias_correction1 = [1 - beta1 ** _get_value(step) for step in grouped_state_steps]
-        unrect_step_size = _stack_if_compiling([(lr * rect / bc) * -1 for rect, bc in zip(unrectified, bias_correction1)])
+        unrect_step_size = [(lr * _get_value(rect) / bc) * -1 for rect, bc in zip(unrectified, bias_correction1)]
         bias_correction2_sqrt_times_rect_step_size = [
-            _dispatch_sqrt(1 - beta2 ** _get_value(step)) * (lr * rect / bc) * -1
+            _dispatch_sqrt(1 - beta2 ** _get_value(step)) * (lr * _get_value(rect) / bc) * -1
```

Comment:
We should probably have a "capturable" code path similar to Adam/NAdam and now Adagrad (https://github.com/pytorch/pytorch/pull/110339) / Adamax

It can use `foreach_sqrt` for the GPU path? :thinking: 

The only reason we wouldn't use it here is because we want this bias correction to be on the Scalar codepath, but that doesn't really make that much sense since it is consumed in foreach below.

Reply:
hmm, I think the issue with converting to foreach here is the `_get_value` part, but honestly I could do list comprehension to create the corresponding step values.

Reply:
Adding a capturable code path may be the right move here. Then, we could get rid of the need for get_value as the capturable path should use the foreach ops, and the eager path can still keep scalar math on python.